pkg/pillar/cmd/volumemgr/initialvolumestatus.go

```diff
@@ -28,61 +28,54 @@ import (
 var nilUUID = uuid.UUID{}
 
 // appRwVolumeName - Returns name of the image ( including parent dir )
-// Note that we still use the sha in the filename to not impact running images. Otherwise
-// we could switch this to imageID
-// XXX other types of volumes might want a different name.
-func appRwVolumeName(sha256, uuidStr string, purgeCounter uint32, format zconfig.Format,
-	origin types.OriginType, isContainer bool) string {
+func appRwVolumeName(uuidStr string, generationCounter uint32,
+	format zconfig.Format, isContainer bool) string {
 
 	purgeString := ""
-	if purgeCounter != 0 {
-		purgeString = fmt.Sprintf("#%d", purgeCounter)
+	if generationCounter != 0 {
+		purgeString = fmt.Sprintf("#%d", generationCounter)
```

Comment:
With the new names we can include #<generationCounter> even when it is zero.

Reply:
Resolved.